makefile

```diff
@@ -1,19 +1,84 @@
-format:
+CONVERTED_NOTEBOOKS_TEMP_DIR := tests/_temp/notebooks
+NOTEBOOKS_DIR := notebooks
+ROOT_RELATIVE_TO_NOTEBOOKS := $(shell realpath --relative-to=$(NOTEBOOKS_DIR) .)
+
+
+.PHONY: default
+default: help
+
+.PHONY: lint
+lint: clean
+	@echo "run linting: mypy"
+	python -m mypy --config-file pyproject.toml maze_transformer/
+	python -m mypy --config-file pyproject.toml tests/
+
+.PHONY: format
+format: clean
+	@echo "run formatting: pycln, isort, and black"
 	poetry run python -m pycln --all .
 	poetry run python -m isort format .
 	poetry run python -m black .
 
-check-format:
+.PHONY: check-format
+check-format: clean
+	@echo "check formatting"
 	poetry run python -m pycln --check --all .
 	poetry run python -m isort --check-only .
 	poetry run python -m black --check .
 
+.PHONY: unit
 unit:
-	rm -rf .pytest_cache
+	@echo "run unit tests"
 	poetry run python -m pytest tests/unit
 
+.PHONY: integration
 integration:
-	rm -rf .pytest_cache
+	@echo "run integration tests"
 	poetry run python -m pytest -s tests/integration
 
-test: unit integration
+.PHONY: convert_notebooks
+convert_notebooks:
+	@echo "convert notebooks in $(NOTEBOOKS_DIR) using tests/helpers/convert_ipynb_to_script.py"
+	python tests/helpers/convert_ipynb_to_script.py notebooks/ --output_dir $(CONVERTED_NOTEBOOKS_TEMP_DIR) --disable_plots
+
+.PHONY: test_notebooks
+test_notebooks: convert_notebooks
+	@echo "test notebooks in $(NOTEBOOKS_DIR)"
+	@echo "Testing notebooks in $(CONVERTED_NOTEBOOKS_TEMP_DIR)"
+	@for file in $(CONVERTED_NOTEBOOKS_TEMP_DIR)/*.py; do \
```

Comment:
Not a huge deal, but I wonder if we should put this logic in a python script to simplify the makefile.

Reply:
Yeah, that would probably make sense! bash is messy and doubly so when in a makefile. I'll look into this.